h/views/activity.py

```diff
@@ -42,9 +42,81 @@ def search(request):
     # Fetch results
     result = query.execute(request, q, page_size=page_size)
 
+    for user in result.aggregations.get('users', []):
+        user['username'] = util.user.split_user(user['user'])['username']
+        del user['user']
+
     return {
         'total': result.total,
         'aggregations': result.aggregations,
         'timeframes': result.timeframes,
         'page': paginate(request, result.total, page_size=page_size),
     }
+
+
+@view_config(route_name='activity.group_search',
+             request_method='GET',
+             renderer='h:templates/activity/search.html.jinja2')
+def group_search(request):
+    if not request.feature('search_page'):
+        raise httpexceptions.HTTPNotFound()
+
+    result = search(request)
+
+    pubid = request.matchdict['pubid']
+
+    try:
+        group = request.db.query(models.Group).filter_by(pubid=pubid).one()
+    except exc.NoResultFound:
+        return result
```

Comment:
We don't want to 404 if the group that the user searches for does not exist (or if it does exist but the user is a member of it). Anyone can type `group:foo<Enter>` into the search box, and an empty search results page seems better than 404ing them.